src/client/client.go

```diff
@@ -232,6 +239,14 @@ func WithDialTimeout(t time.Duration) Option {
 	}
 }
 
+// WithGZIPCompression enabled GZIP compression for data on the wire
+func WithGZIPCompression(enabled bool) Option {
```

Comment:
I think you can remove the argument here, since calling `WithGZIPCompression(false)` is the same as not passing the `Option` at all right?